src/subcommand/merge_subcommand.cpp

```diff
@@ -54,6 +57,56 @@ void perform_fastforward(repository_wrapper& repo, const git_oid target_oid, int
     target_ref.write_new_ref(target_oid);
 }
 
+void create_merge_commit(repository_wrapper& repo, const index_wrapper& index, std::vector<std::string> m_branches_to_merge,
+    const annotated_commit_list_wrapper& commits_to_merge, size_t num_commits_to_merge)
```

Comment:
`create_merge_commit` could be a method of the merge_subcommand class; this way it would have access to the `m_changes_to_merge`  member, and this parameter can be removed. If we prefer to keep it as a free function, then the parameter should be renamed in `branches_to_merge` (the `m_` prefix should be reserved for class members).